core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDetailInfo.java

```diff
@@ -51,7 +51,8 @@
 
   private short versionNumber;
 
-  private short blockletId;
+  // default blockletId should be -1,which means consider all the blocklets in block
```

Comment:
I think it's better not to change the default behavior. Instead we should optimize the behavior of SDK reader for the spark file format.

Reply:
@xuchuanyin ,default  cache level is Block and for block we always set blockletid as -1. so it is better to handle during initialization rather than handle in each reader (if any new reader implemented in future then it will be taken care automatically otherwise each reader should take care same ).